### PR TITLE
Module 12

### PR DESCRIPTION
**Pull request by waormd**

### CODE REVIEW

**Review by JackMead — APPROVED** [review]
This looks good, I can see that Travis has successfully deployed (or at least updated) your infrastructure on Azure. I've left some thoughts below, and in particular we should update the Readme documentation to mention that this is happening and mention any dependencies/workflows for future developers, but otherwise happy for you to merge this when you're ready

File: .gitignore [hunk 1]
```diff
@@ -122,4 +122,40 @@ geckodriver.log
...
+# !example_override.tf
+
+# Include tfplan files to ignore the plan output of command: terraform plan -out=tfplan
+# example: *tfplan*
+
+# Ignore CLI configuration files
+.terraformrc
+terraform.rc
+
+.terraform.lock.hcl
```
[the remark this answers is not in the document]
Comment (waormd) [reply .gitignore]: c44e95abce3434db7b66629b60324e376f875593

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+variable "resource_group_name" {
+    description = "Name of the resource group"
+    default     = "CreditSuisse1_WilliamCole-Jordan_ProjectExercise"
+}
+
+variable "oauth_callback" {}
+variable "oauth_client_id" {}
+variable "oauth_client_secret" {}
+variable "secret_key" {}
```
[the remark this answers is not in the document]
Comment (waormd) [reply variables.tf]: c44e95abce3434db7b66629b60324e376f875593

File: .travis.yml [hunk 3]
```diff
@@ -8,6 +8,16 @@ env:
...
   - secure: Hmo7nllfR15JDSGiKvemph4hmB6R/MfCb7MpM1KQKR0dYa27d84bDMCep/Ee3IVh8FvsI9EMWuWwbJ4w+pBgEPWkLnJlBD7LObbEzMK8NjLf2ciRkNezOlTlfTOJ2pWVIQYDDYMCxJVBdELBOox+kYQRoRQr9npz8wsmHAt3mpdCrdqK+Acte25neL7MPAUrQi+Ql8sdH3E6wBGYgE9j6YvVjKwp1OpMIUvMjzZjGwzJvOnhuJRHK0ms2QOFJh5B6Z4RYBBepftN7kYh9mkvHG8yU7OgVPJhwX2pRo83Nu5Q8K7PVEbPU1K22xVlmfVKU6zfqSSXuuoaZFLqR9Gcs1+R7RgGvom2LeDEtdL+qhWtYJJP6iXhYrUcq6KGQblGDB9jKH2YUcy5BZ5Bynbyahv6zXOKG9f4zQMxI40ELU08vkOKrV2sBPzhLqnjvtTcCrTuGX6bLu8AyRHoxa7gGZ7r1MOmEvVhVPpc1RAhqWcYSvnkUxaLM9gmRMJ1UlE9RI2ZZ7mwecU2mBRedJhaun+18Io5MIsyf/VF4xQA8sRqD5QO3jOx0VdA9HzVA6++rYFTaaGcDiFkureIWfjRcC7QGfF+tpvZjUrctXb9rVi9OKHt2D+cGGBe4G08Ri1EFkX5XbAfeIZJWVKgDxfeaQKtnbqH/RZ4AH+73yIN0Ow=
 
+install:
+  - wget https://releases.hashicorp.com/terraform/"$TF_VERSION"/terraform_"$TF_VERSION"_linux_amd64.zip
+  - unzip terraform_"$TF_VERSION"_linux_amd64.zip
+  - sudo mv terraform /usr/local/bin/
+  - rm terraform_"$TF_VERSION"_linux_amd64.zip
+  - terraform init
```
[the remark this answers is not in the document]
Comment (waormd) [reply .travis.yml]: c44e95abce3434db7b66629b60324e376f875593

File: .travis.yml [hunk 4]
```diff
@@ -8,6 +8,16 @@ env:
...
+
+before_deploy:
```
[the remark this answers is not in the document]
Comment (waormd) [reply .travis.yml]: c44e95abce3434db7b66629b60324e376f875593

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+resource "azurerm_cosmosdb_mongo_database" "db" {
+    name                = "todo-app-cosmos-db-instance"
+    resource_group_name = var.resource_group_name
+    account_name        = azurerm_cosmosdb_account.db.name
+}
+
+resource "azurerm_app_service_plan" "main" {  
+    name                = "terraformed-asp"  
+    resource_group_name = var.resource_group_name  
+    location            = "uksouth"
```
[the remark this answers is not in the document]
Comment (waormd) [reply main.tf]: c44e95abce3434db7b66629b60324e376f875593

**Pull request merged**